pkg/common/plugin/sshpop/handshake.go

```diff
@@ -139,17 +142,24 @@ func (s *ServerHandshake) VerifyAttestationData(data []byte) error {
 	if err := s.s.certChecker.CheckHostKey(addr, &net.IPAddr{}, cert); err != nil {
 		return Errorf("failed to check host key: %v", err)
 	}
+	s.hostname, err = decanonicalizeHostname(cert.ValidPrincipals[0], s.s.canonicalDomain)
+	if err != nil {
+		return Errorf("failed to decanoncalize hostname: %v", err)
+	}
 	s.cert = cert
-	s.hostname = decanonicalizeHostname(cert.ValidPrincipals[0], s.s.canonicalDomain)
 	s.state = stateAttestationDataVerified
 	return nil
 }
 
-func decanonicalizeHostname(fqdn, domain string) string {
+func decanonicalizeHostname(fqdn, domain string) (string, error) {
 	if domain == "" {
-		return fqdn
+		return fqdn, nil
+	}
+	suffix := "." + domain
+	if !strings.HasSuffix(fqdn, suffix) {
```

Comment:
total nitpick (feel free to ignore) but the result from TrimSuffix can also be used to determine if it had the suffix (if not, the result is unchanged):

```
result := strings.TrimSuffix(fqdn, "."+domain)
if result == fqdn {
    return "", fmt.Errorf(....)
}
return result, nil
```